external-crates/move/crates/move-compiler/src/cfgir/liveness/mod.rs

```diff
@@ -288,26 +290,33 @@ mod last_usage {
                 if !context.next_live.contains(v) {
                     match display_var(v.value()) {
                         DisplayVar::Tmp => (),
-                        DisplayVar::Orig(v_str) => {
+                        DisplayVar::Orig(vstr) => {
                             if !v.starts_with_underscore() {
+                                let case_msg = match case {
+                                    AssignCase::Update => "assignment for ",
+                                    AssignCase::Let => "",
+                                };
                                 let msg = format!(
-                                    "Unused assignment for variable '{}'. Consider \
+                                    "Unused {case_msg}variable '{vstr}'. Consider \
                                      removing, replacing with '_', or prefixing with '_' (e.g., \
-                                     '_{}')",
-                                    v_str, v_str
+                                     '_{vstr}')",
                                 );
                                 context
                                     .env
                                     .add_diag(diag!(UnusedItem::Assignment, (l.loc, msg)));
                             }
-                            if context.has_drop(v) {
+                            let should_ignore =
+                                v.starts_with_underscore() || case == AssignCase::Update;
```

Comment:
A comment explaining the logic here may be nice: if this is an update, we know the variable isn't live after this, and so this assignment is subject to erasure.